perun-core/src/main/java/cz/metacentrum/perun/core/impl/Auditer.java

```diff
@@ -537,11 +536,52 @@ public void setLastProcessedId(String consumerName, int lastProcessedId) throws
 	}
 
 	/**
+	 * Stores the auditer messages to the DB in batch.
+	 *
+	 * @param messages list of AuditerMessages
+	 */
+	public void storeMessagesToDb(final List<AuditerMessage> messages) {
+		synchronized (LOCK_DB_TABLE_AUDITER_LOG) {
+			try {
+				jdbc.batchUpdate("insert into auditer_log (id, msg, actor, created_at, created_by_uid) values (?,?,?," + Compatibility.getSysdate() + ",?)",
+						new BatchPreparedStatementSetter() {
+							@Override
+							public void setValues(PreparedStatement ps, int i) throws SQLException {
+								final AuditerMessage auditerMessage = messages.get(i);
+								final String message = auditerMessage.getMessage();
+								final PerunSession session = auditerMessage.getOriginaterPerunSession();
+								log.info("AUDIT: {}", message);
+								try {
+									ps.setInt(1, Utils.getNewId(jdbc, "auditer_log_id_seq"));
+								} catch (InternalErrorException e) {
+									log.error("Cannot get unique id for new auditer log message ['{}'], exception: {}", message, e);
```

Comment:
Consider pass the exception up the calling chain. Without ID, the SQL query will fail later anyway.